sherlock.h

```diff
@@ -89,18 +90,63 @@
 
 namespace sherlock {
 
-// TODO(dkorolev): Test this code.
+template <typename E>
+struct ExtractTimestampImpl {
+  static bricks::time::EPOCH_MILLISECONDS ExtractTimestamp(const E& entry) { return entry.ExtractTimestamp(); }
+};
+
+template <typename E>
+struct ExtractTimestampImpl<std::unique_ptr<E>> {
+  static bricks::time::EPOCH_MILLISECONDS ExtractTimestamp(const std::unique_ptr<E>& entry) {
+    return entry->ExtractTimestamp();
+  }
+};
+
+template <typename E>
+bricks::time::EPOCH_MILLISECONDS ExtractTimestamp(E&& entry) {
+  return ExtractTimestampImpl<typename std::remove_reference<E>::type>::ExtractTimestamp(
+      std::forward<E>(entry));
+}
+
 template <typename E>
 class PubSubHTTPEndpoint {
  public:
   PubSubHTTPEndpoint(const std::string& value_name, Request r)
       : value_name_(value_name),
-        http_request_scope_(std::move(r)),
-        http_response_(http_request_scope_.SendChunkedResponse()) {}
+        http_request_(std::move(r)),
+        http_response_(http_request_.SendChunkedResponse()) {
+    // TODO(dkorolev): `"from"`, etc.
+    if (http_request_.url.query.has("recent")) {
+      serving_ = false;
+      from_timestamp_ =
+          r.timestamp - static_cast<bricks::time::MILLISECONDS_INTERVAL>(
```

Comment:
This looks for entries recent relative to the time when the request has reached the serving backend. It would be more accurate if we accept the timestamp from the client.